### DISCUSSION TITLE
OpenID logout

### DISCUSSION THREAD

**Discussion by meedkruu** [directus/directus#5487] [Feature Requests]
As my colleague posted already yesterday on discord, we are using OpenID connect for authentication. To fully comply with the OpenID provider we use, we would need to have a general logout using the OpenID provider as described in the specification '[OpenID Connect RP-Initiated Logout 1.0](https://openid.net/specs/openid-connect-rpinitiated-1_0.html)'.
The OpenID provider provides an end session endpoint. I would suggest, that this becomes another OAUTH environment variable. When this is given, directus should redirect to that endpoint instead of doing just a local logout. The OpenID provider will then call the logout URL of directus, which has to be registered at the OpenID provider.
This feature is very important for us, we would very much appreciate, if this could be implemented...

**Comment by rijkvanzanten** [comment]
Directus doesn't currently keep the session alive with the provider in the first place, so it won't be trivially easy to add this in 🤔 Right now, we only use the oAuth flow (or OpenID based on configured scope) to check if the user exists and is authorized in the external service provider, after which the session is handled on the Directus side.

> This feature is very important for us, we would very much appreciate, if this could be implemented...

You could look into making this a commissioned feature if it's crucial for your use case. See https://docs.directus.io/getting-started/support/#commissioned-features for more information.

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
